File: CSharp part2/01. Arrays/05.MaximalIncreasingSequence/MaximalIncreasingSequence.cs
// Write a program that finds the maximal increasing sequence in an array.

using System;

class MaximalIncreasingSequence
{
    static void Main()
    {
        Console.Write("Enter array elements(separated by a space):");
        string number = Console.ReadLine();
        string[] splitNumbers = number.Split(' ');
        int[] arr = new int[splitNumbers.Length];
        for (int i = 0; i < arr.Length; i++)
        {
            arr[i] = int.Parse(splitNumbers[i]);
        }
        int maxstart = 0;
        int max = 1;
        int start = 0;
        for (int i = 1; i < arr.Length; i++) 
        {
            if (arr[i] > arr[i - 1]) 
            {
               if (i - start + 1 > max) 
               {
                   max = i - start + 1;
                   maxstart = start;
               }
            } 
            else 
            {
               start = i;
            }
        }
        for (int i = 0; i < max; i++)
        {
            Console.Write(arr[maxstart+i]+", ");
        }
        Console.WriteLine();
    }   
}

File: JavaScript-Fundamentals/Arrays/04.MaximalIncreasingSequence/MaximalIncreasingSequence.js
// Write a script that finds the maximal increasing sequence in an array.

var jsConsole;
var btn = document.getElementById('findMaxIncreasingSequence');
btn.onclick = findMaxIncreasingSequence;

function findMaxIncreasingSequence () {
	var array = jsConsole.read('#array').split(" ");

	var maxstart = 0;
	var max = 1;
	var start = 0;

	for (var i = 0; i < array.length; i++) {
		if (array[i] > array[i - 1]) {
			if (i - start + 1 > max) {
				max = i - start + 1;
				maxstart = start;
			}
		} else {
			start = i;
		}
	}

	for (var j = 0; j < max; j++) {
		jsConsole.write(array[maxstart + j] + " ");
		console.log(array[maxstart + j] + " ");
	}
}